graphql/schema/rules.go

```diff
@@ -47,7 +47,11 @@ func init() {
 	validator.AddRule("Check variable type is correct", variableTypeCheck)
 	validator.AddRule("Check arguments of cascade directive", directiveArgumentsCheck)
 	validator.AddRule("Check range for Int type", intRangeCheck)
-	validator.AddRule("Input Coercion to List", listInputCoercion)
+	// Graphql accept both single object and array of objects as value when the schema is defined
+	// as an array. listInputCoercion changes the value to array if the single object is provided.
+	// Changing the value can mess up with the other data validation rules hence we are setting
+	// up the order to a high value so that it will be executed last.
+	validator.AddRuleWithOrder("Input Coercion to List", 100, listInputCoercion)
```

Comment:
We should create a GitHub issue for fixing this correctly. Number 100 should not hardcoded. What if we have more than 100 rules?

Reply:
Limiting is a good practice, but also left a note for the user is good.

If you don't limit some things, abuse can happen. Especially in shared clusters. Some user might run a very wide complex query and bring down a cluster.

Reply:
@MichelDiz The 100 value is not a size limit on the array, rather the _position_ of the `rule` in the list. At present there are only 4 other rules in the gqlparser validator, so 100 definitely places it at the bottom.

Ideally the gqlparser's API would allow access to the length of the list, so one could make a dynamic decision, but at present that data structure to private to the package.

I don't like seeing hardcoded stuff like this either, but in this case it's not as bad as it looks.

Reply:
@matthewmcneely what are your thoughts here https://github.com/dgraph-io/dgraph/pull/8320#pullrequestreview-1121706742 ... I think we should do it right.

I understand there are only 4 other rules in the gqlparser, but the number of rules is capped by MaxInt value. Can we pin this to `MaxInt`. This way we are also being explicit (vs. implicit) in the code.

Reply:
Maybe MaxInt isn't the best solution... what was wanted is that for that rule to be at the end _of where the set of rules currently stands_, MaxInt definitely does that but what about the next Rule that someone adds that wants to go just before?

An alternative: create a few consts that can be used to place items in logical order, say

```
const (
    baseRules = 0
    listCoercionRules = 1000
    fooRules = 2000
)
```

With this, subsequent authors can decide where in the logical order they want their rules to go.